File: javascript/raindrops/raindrops.js
// @ts-check

const NOISES = Object.freeze({ 3: 'Pling', 5: 'Plang', 7: 'Plong' });

/** @param {number} n */
export function convert(n, s='') {
  for (const prime in NOISES) n % +prime || (s += NOISES[prime]);
  return s || s + n; };

File: fortran/raindrops/raindrops.f90
MODULE raindrops; CONTAINS
  PURE CHARACTER*15 FUNCTION convert(n) RESULT(s); INTENT(IN) n

    INTEGER(1), PARAMETER :: PRIMES(3) = (/3_1, 5_1, 7_1/)
    CHARACTER, PARAMETER :: NOISES(3)*5 = (/"Pling", "Plang", "Plong"/)

    s = ""; DO i = 1, size(PRIMES)
      IF (mod(n, int(PRIMES(i))) == 0) s(1 + len_trim(s):) = NOISES(i); END DO

    IF (s .EQ. '') WRITE (s, '(I0)') n; END FUNCTION; END MODULE
